includes/payment-tokens/class-wc-stripe-klarna-payment-token.php

```diff
@@ -37,54 +37,74 @@ class WC_Payment_Token_Klarna extends WC_Payment_Token implements WC_Stripe_Paym
 	 * @return string The name of the token to display
 	 */
 	public function get_display_name( $deprecated = '' ) {
-		$dob = $this->get_dob();
-
-		if ( empty( $dob ) ) {
+		if ( empty( $this->get_dob() ) ) {
 			return __( 'Klarna', 'woocommerce-gateway-stripe' );
 		}
 
-		// Translators: %s is the customer's date of birth.
-		return sprintf( __( 'Klarna (%s)', 'woocommerce-gateway-stripe' ), $this->format_dob( $dob ) );
+		// Translators: %s is the customer's masked date of birth.
+		return sprintf( __( 'Klarna (%s)', 'woocommerce-gateway-stripe' ), $this->get_masked_dob() );
 	}
 
 	/**
 	 * Sets the Klarna token's date of birth.
 	 *
-	 * @param object $dob The date of birth.
+	 * @param string $dob The formated date of birth object (YYYY-mm-dd).
 	 */
 	public function set_dob( $dob ) {
 		$this->set_prop( 'dob', $dob );
 	}
 
 	/**
-	 * Fetches Klarna token's date of birth.
+	 * Fetches Klarna token's date of birth (formated to Y-m-d).
 	 *
-	 * @return object The Klarna token's date of birth.
+	 * @return string The Klarna token's date of birth.
 	 */
 	public function get_dob() {
 		return $this->get_prop( 'dob' );
 	}
 
+	/**
+	 * Formats and sets the Klarna token's date of birth.
+	 *
+	 * @param object $dob The raw date of birth object (from Stripe's API).
+	 */
+	public function set_formated_dob( $dob ) {
+		$this->set_dob( $this->format_dob( $dob ) );
+	}
+
+	/**
+	 * Returns the masked date of birth for display.
+	 *
+	 * @return string
+	 */
+	public function get_masked_dob() {
+		if ( empty( $this->get_dob() ) ) {
+			return '';
+		}
+
+		return 'xxxx' . gmdate( '-m-d', strtotime( $this->get_dob() ) );
+	}
+
 	/**
 	 * Checks if the payment method token is equal a provided payment method.
 	 *
 	 * @inheritDoc
 	 */
 	public function is_equal_payment_method( $payment_method ): bool {
-		if ( WC_Stripe_Payment_Methods::KLARNA === $this->get_type() ) {
-			$method_dob = $payment_method->klarna->dob ?? null;
-			if ( null === $method_dob && null === $this->get_dob() ) {
-				return true;
-			}
-
-			$formated_method_dob = $this->format_dob( $method_dob );
-			$formated_dob        = $this->format_dob( $this->get_dob() );
-			if ( $formated_method_dob === $formated_dob ) {
-				return true;
-			}
+		if ( WC_Stripe_Payment_Methods::KLARNA !== $this->get_type() ) {
```

Comment:
Should we not be checking `$payment_method->get_type()` here? As written, this condition will never be false.
```suggestion
		if ( WC_Stripe_Payment_Methods::KLARNA !== $payment_method->get_type() ) {
```

Reply:
Oops! Fixed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4658/commits/a42448167043cc1dd2b1aa16259c855340fc38d0 (sorry, I did not see your code block suggesiton)

Reply:
Looking at the test failures, `$payment_method` is a basic object, so this access actually needs to be `$payment_method->type`. I fixed that in aa39d909a31781352bfe92ec3f971054834a2b27.